process/energy.go

```diff
@@ -0,0 +1,167 @@
+package process
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-accounts-manager/core"
+	"github.com/ElrondNetwork/elrond-accounts-manager/data"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	hexEncodedEnergyPrefix = "75736572456e65726779"
+)
+
+// GetAccountsWithEnergy will return accounts with energy
+func (ag *accountsGetter) GetAccountsWithEnergy(currentEpoch uint32) (map[string]*data.AccountInfoWithStakeValues, error) {
+	if ag.energyContractAddress == "" {
+		return map[string]*data.AccountInfoWithStakeValues{}, nil
+	}
+
+	genericAPIResponse := &data.GenericAPIResponse{}
+	path := fmt.Sprintf(pathAccountKeys, ag.energyContractAddress)
+	err := ag.restClient.CallGetRestEndPoint(path, genericAPIResponse, core.GetEmptyApiCredentials())
+	if err != nil {
+		return nil, err
+	}
+	if genericAPIResponse.Error != "" {
+		return nil, fmt.Errorf("get accounts with energy %s", genericAPIResponse.Error)
+	}
+
+	return ag.extractAddressesAndEnergy(genericAPIResponse.Data, currentEpoch)
+}
+
+func (ag *accountsGetter) extractAddressesAndEnergy(accountStorage []byte, currentEpoch uint32) (map[string]*data.AccountInfoWithStakeValues, error) {
+	pairs := gjson.Get(string(accountStorage), "data.pairs")
+
+	keyValueMap := make(map[string]string)
+	err := json.Unmarshal([]byte(pairs.String()), &keyValueMap)
+	if err != nil {
+		return nil, fmt.Errorf("cannot unmarshal account storage, error: %s", err.Error())
+	}
+
+	accountsWithEnergy := make(map[string]*data.AccountInfoWithStakeValues)
+	for key, value := range keyValueMap {
+		address, ok := ag.extractAddressFromKey(key)
+		if !ok {
+			continue
+		}
+		energy, ok := ag.extractEnergyFromValue(value)
+		if !ok {
+			continue
+		}
+
+		energyValue := calculateEnergyValueBasedOnCurrentEpoch(energy, currentEpoch)
+
+		// ignore addresses with energyValue less than zero
+		zero := big.NewInt(0)
+		if zero.Cmp(energyValue) > 0 {
+			continue
+		}
+
+		accountsWithEnergy[address] = &data.AccountInfoWithStakeValues{
+			StakeInfo: data.StakeInfo{
+				Energy:    energyValue.String(),
+				EnergyNum: core.ComputeBalanceAsFloat(energyValue.String()),
+			},
+		}
+	}
+
+	return accountsWithEnergy, nil
+}
+
+func (ag *accountsGetter) extractAddressFromKey(key string) (string, bool) {
+	hasPrefix := strings.HasPrefix(key, hexEncodedEnergyPrefix)
+	if !hasPrefix {
+		return "", false
+	}
+
+	hexEncodedAddress := strings.ReplaceAll(key, hexEncodedEnergyPrefix, "")
+	addressBytes, err := hex.DecodeString(hexEncodedAddress)
+	if err != nil {
+		log.Warn("cannot decode address from key", "error", err)
+		return "", false
+	}
+
+	return ag.pubKeyConverter.Encode(addressBytes), true
+}
+
+type energyStruc struct {
```

Comment:
energyStruct ?

Reply:
re-arrange the fields: big ints first 2 positions, and the uint32 at the end

Reply:
renamed and changed the order of the fields